### PR TITLE
[ticket/13799] Do not rawurlencode category in gallery avatar

### PR DESCRIPTION
**Pull request by marc1706**
Otherwise the directory seperator will incorrectly be transformed to its
HTML replacement.

Ticket: https://tracker.phpbb.com/browse/PHPBB3-13799

PHPBB3-13799

### CODE REVIEW

**Comment by bantu** [comment]
I am not sure this is a correct fix. Both $image and $cat are based on data returned from the iterator, but only one is urlencoded post-patch.

**Comment by marc1706** [comment]
So yeah, the selected avatar never showed as selected either. Fixed that now, too.

File: phpBB/styles/prosilver/template/ucp_avatar_options_local.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 <!-- BEGIN avatar_local_row -->
 <!-- BEGIN avatar_local_col -->
 	<label for="av-{avatar_local_row.S_ROW_COUNT}-{avatar_local_row.avatar_local_col.S_ROW_COUNT}"><img src="{avatar_local_row.avatar_local_col.AVATAR_IMAGE}" alt="" /><br />
-	<input type="radio" name="avatar_local_file" id="av-{avatar_local_row.S_ROW_COUNT}-{avatar_local_row.avatar_local_col.S_ROW_COUNT}" value="{avatar_local_row.avatar_local_col.AVATAR_FILE}" /></label>
+	<input type="radio" name="avatar_local_file" id="av-{avatar_local_row.S_ROW_COUNT}-{avatar_local_row.avatar_local_col.S_ROW_COUNT}" value="{avatar_local_row.avatar_local_col.AVATAR_FILE}"<!-- IF avatar_local_row.avatar_local_col.CHECKED -->checked="checked"<!-- ENDIF --> /></label>
```
Comment (nickvergessen) [inline phpBB/styles/prosilver/template/ucp_avatar_options_local.html]: missing space before `checked`
mind breaking the line aswell?

File: phpBB/styles/subsilver2/template/ucp_avatar_options_local.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
 			</tr>
 			<tr>
 				<!-- BEGIN avatar_local_option -->
-					<td class="row2" align="center"><input type="radio" class="radio" name="avatar_local_file" id="av-{avatar_local_row.S_ROW_COUNT}-{avatar_local_row.avatar_local_col.S_ROW_COUNT}" value="{avatar_local_row.avatar_local_option.S_OPTIONS_AVATAR}" /></td>
+					<td class="row2" align="center"><input type="radio" class="radio" name="avatar_local_file" id="av-{avatar_local_row.S_ROW_COUNT}-{avatar_local_row.avatar_local_col.S_ROW_COUNT}" value="{avatar_local_row.avatar_local_option.S_OPTIONS_AVATAR}"<!-- IF avatar_local_row.avatar_local_option.CHECKED -->checked="checked"<!-- ENDIF --> /></td>
```
Comment (nickvergessen) [inline phpBB/styles/subsilver2/template/ucp_avatar_options_local.html]: same here

File: phpBB/phpbb/avatar/driver/local.php [hunk 3]
```diff
@@ -182,7 +184,7 @@ protected function get_avatar_list($user)
 					}
 					$cat = ($path == $file_path) ? $user->lang['NO_AVATAR_CATEGORY'] : str_replace("$path/", '', $file_path);
 					$avatar_list[$cat][$image] = array(
-						'file'      => ($cat != $user->lang['NO_AVATAR_CATEGORY']) ? rawurlencode($cat) . '/' . rawurlencode($image) : rawurlencode($image),
+						'file'      => ($cat != $user->lang['NO_AVATAR_CATEGORY']) ? str_replace('%2F', '/', rawurlencode($cat)) . '/' . rawurlencode($image) : rawurlencode($image),
```
Comment (nickvergessen) [inline phpBB/phpbb/avatar/driver/local.php]: the user can not excape the path now, by specifying `./../../` etc?

Comment (marc1706) [inline phpBB/phpbb/avatar/driver/local.php]: No. The image list is retrieved from the filesystem. The user can only selected those files. Trying to select categories that do not exist in this list or images that do not exist in that list, e..g by using ```../../``` will result in an empty list for the categories and an error message for the image.

**Comment by Nicofuma** [comment]
unrelated faillure

**Pull request merged**